lib/document/templates.js

```diff
@@ -123,18 +124,36 @@ ${faDom.css()}
   return ''
 }
 
+// <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css"/>
+const linkStylesheet = /(<link rel="stylesheet"[^>]*\/>)/g;
+
 const syntaxHighlighterHead = (node, syntaxHighlighter, attrs) => {
+  let header = ''
   if (syntaxHighlighter !== Opal.nil && syntaxHighlighter['$docinfo?']('head')) {
-    return syntaxHighlighter['$docinfo']('head', node, Opal.hash(attrs))
+    header = header + syntaxHighlighter['$docinfo']('head', node, Opal.hash(attrs))
   }
-  return ''
+  if (syntaxHighlighter !== Opal.nil && syntaxHighlighter['$docinfo?']('footer')) {
+    // reordering all link elements from footer to header, as pagedjs would otherwise render them too late in the process
+    let footer = syntaxHighlighter['$docinfo']('footer', node, Opal.hash(attrs))
+    const matches = footer.match(linkStylesheet)
+    if (matches !== null) {
+      matches.forEach((element) => {
+        header = header + element
+      })
+    }
+  }
+  return header
 }
 
 const syntaxHighlighterFooter = (node, syntaxHighlighter, attrs) => {
+  let footer = ''
   if (syntaxHighlighter !== Opal.nil && syntaxHighlighter['$docinfo?']('footer')) {
-    return syntaxHighlighter['$docinfo']('footer', node, Opal.hash(attrs))
+    // skip all link elements in the footer as they are re-ordered to the header by syntaxHighlighterHead()
+    footer = footer + syntaxHighlighter['$docinfo']('footer', node, Opal.hash(attrs))
+    // <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css"/>
+    footer = footer.replace(/<link rel="stylesheet"[^>]*\/>/g, '')
```

Comment:
Should be replace with the constant:

```suggestion
    footer = footer.replace(linkStylesheet, '')
```